internal/controller/numaflowcontrollerrollout/numaflowcontrollerrollout_controller.go

```diff
@@ -269,539 +226,312 @@ func (r *NumaflowControllerRolloutReconciler) reconcile(
-				phase, err := r.sync(controllerRollout, namespace, numaLogger)
-				if err != nil {
-					return err
-				}
-
-				r.customMetrics.ReconciliationDuration.WithLabelValues(ControllerNumaflowControllerRollout, "update").Observe(time.Since(syncStartTime).Seconds())
-				if phase != gitopsSyncCommon.OperationSucceeded {
-					return fmt.Errorf("sync operation is not successful")
-				}
-				return nil
-			},
-			pipelinerollout.PipelineROReconciler.EnqueuePipeline)
-		if err != nil {
-			return ctrl.Result{}, err
+			return ctrl.Result{}, fmt.Errorf("error getting NumaflowController: %v", err)
 		}
-		if !done {
-			return common.DefaultDelayedRequeue, nil
-		}
-
 	}
 
-	// apply controller - this handles syncing in the cases in which our Controller Rollout isn't updating
-	// (note that the cases above in which it is updating have a 'return' statement):
-	// - new ControllerRollout
-	// - auto healing
-	// - somebody changed the manifest associated with the Controller version (shouldn't happen but could)
-	phase, err := r.sync(controllerRollout, namespace, numaLogger)
+	// Object already exists: perform logic related to updating
+	newNumaflowControllerDef = r.merge(existingNumaflowControllerDef, newNumaflowControllerDef)
+	needsRequeue, err := r.processExistingNumaflowController(ctx, nfcRollout, existingNumaflowControllerDef, newNumaflowControllerDef, syncStartTime)
 	if err != nil {
-		return ctrl.Result{}, err
+		return ctrl.Result{}, fmt.Errorf("error processing existing NumaflowController: %v", err)
 	}
-
-	if phase != gitopsSyncCommon.OperationSucceeded {
-		return ctrl.Result{}, fmt.Errorf("sync operation is not successful")
-	}
-
-	// Generate the creation metrics only if the numaflow controller is newly created
-	if !deploymentExists {
-		r.customMetrics.ReconciliationDuration.WithLabelValues(ControllerNumaflowControllerRollout, "create").Observe(time.Since(syncStartTime).Seconds())
+	if needsRequeue {
+		return common.DefaultDelayedRequeue, nil
 	}
 
 	return ctrl.Result{}, nil
 }
 
 // for the purpose of logging
 func (r *NumaflowControllerRolloutReconciler) GetChildTypeString() string {
-	return "Numaflow Controller"
+	return "numaflowcontroller"
 }
 
-func (r *NumaflowControllerRolloutReconciler) GetPipelineList(ctx context.Context, rolloutNamespace string, _ string) (*unstructured.UnstructuredList, error) {
-	return kubernetes.ListLiveResource(ctx, common.NumaflowAPIGroup, common.NumaflowAPIVersion, "pipelines", rolloutNamespace, common.LabelKeyParentRollout, "")
+func (r *NumaflowControllerRolloutReconciler) GetPipelineList(ctx context.Context, rolloutNamespace string, rolloutName string) (*unstructured.UnstructuredList, error) {
+	gvk := schema.GroupVersionKind{Group: common.NumaflowAPIGroup, Version: common.NumaflowAPIVersion, Kind: common.NumaflowPipelineKind}
+	// List all the pipelines since they are all managed by the same single NumaplaneController
+	return kubernetes.ListResources(ctx, r.client, gvk, rolloutNamespace)
```

Comment:
(deleted my original comment here in case you saw it - decided it was not an issue)